.github/workflows/nodejs-macos.yml

```diff
@@ -0,0 +1,26 @@
+name: Node.js (macOS) CI
+
+on:
+  push:
+    branches: [ "master" ]
+  schedule:
+    - cron:  '5 4 * * 0'
+
+jobs:
+  npm-test:
+
+    runs-on: macos-latest
```

Comment:
Not a blocker, but is it possible to use the same workflow for all 3 platforms? Or is it to limit testing the CI to only run the latest node on macOS and windows, and let ubuntu test the spread of supported versions?

The matrix strategy in nodejs-ubuntu.yml supports specifying os too.

Reply:
Can we change this in a subsequent PR? Adding testing multiple node versions on all OSs does increase the pipeline time.

Reply:
https://github.com/hubot-new/hubot/pull/62#issuecomment-1500961898 was the reasoning, no objection to moving that to a common file if Actions will support that syntax. Making all PRs run on Windows would add a few minutes to each pipeline execution, running Ubuntu on PRs was around the 30s mark.

- Windows - 1m45 @ https://github.com/hubot-new/hubot/actions/runs/4648747554
- Ubuntu - 48s @ https://github.com/hubot-new/hubot/actions/runs/4648748957
- MacOS - 54s @ https://github.com/hubot-new/hubot/actions/runs/4648746939

Hardly a killer wait, but enough that when I turned it on, I didn't want to double (or more, with a matrix) test run time for what I felt was likely not a huge increase in coverage. There was one code location (hubot-new/hubot@3743c7fc8ecb709422290f50837f685d0990af7a) where OS-specific code was affected, but we're also validating that dependencies work on various OS.

Reply:
Will leave this PR as-is for now.

I've opened https://github.com/hubotio/hubot/issues/1582 to document this change in follow-up and propose removing the TravisCI configuration. (I don't have anything against TravisCI, but it's one more thing to maintain and Github Actions does what we need IMO.)

I agree that a single workflow configuration would be clearer; if we _have_ to use multiple then the yaml should have a comment explaining why.